### ISSUE TITLE
Can't add bag to store

### ISSUE THREAD

**Issue by simonpang** [siuying/NanoStoreInMotion#9]
I can't make Bag to work. Seems it can save to sqlite but can't get back the objects. I can't pass this case:

``` ruby
  it "should add bag to store" do

    before_count = NanoStore.shared_store.bags.count

    bag = Bag.bag
    NanoStore.shared_store.addObject(bag, error:nil)

    # use << method to add object to bag
    page = Page.new
    page.text = "Hello"
    page.index = 1
    bag << page 

    bag.save
    NanoStore.shared_store.save   #PS: is this needed?

    NanoStore.shared_store.bags.count.should.be == before_count + 1
  end
```

**Comment by siuying** [comment]
i confirm the issue exists. weird following case work:

``` ruby
retrievedBag = NanoStore.shared_store.bagWithName(bag.name)
retrievedBag.name.should.be == bag.name
```

Would see can I reproduce this in ObjC. 

I really didn't used bags features in my app so it would need more attention ...

**Comment by naoyamakino** [comment]
I encountered the same issue, @siuying do you have an update on this?
also I was able to retrieve a bag using `bagWithName` but after I killed the app and restart again, I was not able to retrieve a bag with the same name (I use persistent storage for NanoStore). I guess a bag is not persistent, is it?
Is there any way we can create persistent bags?

**Comment by siuying** [comment]
This is caused by NanoStore query bags with class name "NSFNanoBag", and we're using its subclass (NanoStore::Bag) in motion. Fixes in commit c431027
